proj/src/main/webapp/script.js

```diff
@@ -3,48 +3,69 @@
  */
 
 // Elements of login container
+const fname = document.getElementById("fname")
 const txtEmail = document.getElementById("email");
 const txtPassword = document.getElementById("pass");
 const btnLogin = document.getElementById("btnLogin");
 const btnSignUp = document.getElementById("btnSignUp");
 const btnLogout = document.getElementById("btnLogout");
 
 // Add login event
-btnLogin.addEventListener("click", e => {
-    const emailVal = txtEmail.value;
-    const passVal = pass.value;
+if(btnLogin){
+    btnLogin.addEventListener("click", e => {
+        const emailVal = txtEmail.value;
+        const passVal = pass.value;
 
-    // Initialize auth object
-    const auth = firebase.auth();
+        // Initialize auth object
+        const auth = firebase.auth();
 
-    const promise = auth.signInWithEmailAndPassword(emailVal, passVal);
-    promise.catch(e => console.log(e.message));
-});
-
-// Add sign up event 
-btnSignUp.addEventListener("click", e => {
-    const emailVal = txtEmail.value;
-    const passVal = pass.value;
+        const promise = auth.signInWithEmailAndPassword(emailVal, passVal).then(function(user){
+            window.location.replace("chat.html");
+        });
+        promise.catch(e => console.log(e.message));
+    });
+}
 
-    // Initialize auth object
-    const auth = firebase.auth();
+// Add sign up event
+if(btnSignUp){
+    btnSignUp.addEventListener("click", e => {
+        const emailVal = txtEmail.value;
+        const passVal = pass.value;
+
+        // Initialize auth object
+        const auth = firebase.auth();
+        auth.useDeviceLanguage();
+
+        const promise = auth.createUserWithEmailAndPassword(emailVal, passVal).then(function(){
+            auth.currentUser.updateProfile({displayName: fname.value + " " + lname.value}).then(function(){
```

Comment:
here i think you forgot to initialize lname because i don't see it anywhere else in this file